spec/models/miq_group_spec.rb

```diff
@@ -255,6 +255,20 @@
       expect { expect { group.destroy }.to raise_error(RuntimeError, /The group has users assigned that do not belong to any other group/) }.to_not change { MiqGroup.count }
     end
 
+    it "fails if the group is the current_user group for any user in the group" do
+      group2 = FactoryGirl.create(:miq_group)
+      FactoryGirl.create(:user, :miq_groups => [group, group2], :current_group => group2)
+      FactoryGirl.create(:user, :miq_groups => [group, group2], :current_group => group)
+      expect { expect { group.destroy }.to raise_error(RuntimeError, /The group is the login group for at least one user in the group/) }.to_not change { MiqGroup.count }
```

Comment:
Shouldn't succeed because both users are members of another group and you new logic would switch to current_group to `group2` for the second user?